webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/core_menu_options.tsx

```diff
@@ -32,6 +37,36 @@ function CoreMenuOptions({handleOnSelect}: Props) {
         />
     );
 
+    const extraProps: Partial<MenuItemProps> = {};
+
+    if (isDM) {
+        function getScheduledTimeInTeammateTimezone(userCurrentTimestamp: number, teammateTimezoneString: string): string {
```

Comment:
Since its a static function, you can simply declare it outside the component. That way, its not re-declared on multiple renders.

Reply:
will be modified here: https://github.com/mattermost/mattermost/pull/29120/files#diff-3d804cbb710ab6df92df20bc7c511ba94dc7cbe9a8060150aa1cf03c49297ee9R41